Werkgroep API design rules/naamgeving.md

```diff
@@ -0,0 +1,56 @@
+# Naamgeving
+## Generieke naamconventies
+### Speciale tekens
+- Diakrieten normaliseren naar basis-karakter
+- Symbolen/leestekens (met uitzondering van een koppelteken) weglaten
+ 
+## Resource namen en path segments
+Omdat resources dingen beschrijven (in plaats van acties), worden resource namen als zelfstandig naamwoorden beschreven. Enkelvoudige resources worden in enkelvoud geschreven, collection resources in meervoud:
+(etc--kan voortborduren op 3.1 Resources van de API Design Rules).
+
+In het geval van samengestelde resource namen of path segments, wordt de kebab-case casing conventie gehanteerd.
+
+### Voorbeelden:
+- `/gebouw`
+- `/vergunningen`
+- `/organisatie-codes`
+ 
+### Niet doen:
+- Gebruik geen operaties zoals gebruikelijk is bij RPC-style API's (bv /getResource).
+- Refereer niet in de naam dat iets een API is (bv /resource_api).
+- Gebruik geen bestands extensies in de resource namen (bv /resource.json).
+    - Het gebruik van Content Negotiation via de Accept header heeft de voorkeur, al zijn API aanbieders vrij om convenience endpoints aan te bieden op basis van extensies.
+- Voorkom waar mogelijk het gebruik van afkortingen voor concepten, zoals `/coords` ipv `/coordinaten`. Een uitzondering hierop is het gebruik van breed bekende afkortingen als kwalificatie in veldnamen zoals `bagNummeraanduiding`. 
+- Gebruik geen samengestelde namen voor geneste objecten (bv /adres_straat).
+    - Gebruik hiervoor een sub-object 'adres' met daarin een 'straat' parameter.
+
+## Resource bodies
+- Gebruik JSON of JSON gebaseerde documenttypen (bv JSON+HAL)
+- Gebruik camelCase elementen
+
+## Request parameters
+Het voorstel voor request parameters (query parameters en in de request body) is:
+
+- Technische request parameters (gebruikt om te filteren, sorteren, etc) bij voorkeur Engelstalig (`sort`, `filter`, `expand` etc);
+- GEEN gebruik van underscores (dus `sort` ipv `_sort`), omdat het anders zal betekenen dat (bijna) alle request parameters met een underscore worden geschreven.
```

Comment:
Het afraden lijkt me sowieso niet handig want het strookt niet met de DSO 2.0 specificatie. Het afdwingen wellicht ook weer niet. 

Ik zou voorstellen alleen op de te nemen dat *wanneer* velden geprefixed worden met een underscore dit geinterpreteerd moet worden als een indicatie dat het gaat om een veld op meta-niveau. (i.e. MAY be prefixed with underscore to indicate that ....)

Dat is consistent met de API standaarden die er nu zijn en helpt toch sturing en structuring te geven waar gewenst.

Reply:
Dank je @mtrimpe . We hebben dit tijdens de werkgroep besproken en besloten om voor parameters zonder underscore te gaan. Dat sommige keuzen bij DSO anders zijn gemaakt kunnen we volgens mij niet voorkomen, tenzij we de DSO specificatie kopiëren en als NL API Strategie gebruiken..

Reply:
Ik heb overigens het document waarnaar je refereert leeggemaakt, omdat dat een werkdocument betrof en de te reviewen teksten in de ADR extensies zelf zijn opgenomen.

Reply:
We hebben in alle standaardvorming tot nu toe gekozen om maar zeer beperkt incongruenties met significante bestaande Nederlandse standaarden op te nemen en waar dit wel is gebeurd is dat pas na significant overleg en zwaarwegend belang gedaan.

Ik zie ook geen/zeer beperkte functionele meerwaarde van het afdwingen van het *niet* hebben van underscores terwijl het wel zeer significante problemen introduceert voor API's met een breed aanbod aan vaak extern gegeven datamodellen. Deze beperking je dwingt om elke query parameter die het gedrag van de API wijzigt (page, pageSize, format, fields, expand, embed, etc. etc. etc.) als reserved/forbidden field-name in je datastructuur te definieren.